plugins/CoreAdminHome/SystemSettings.php

```diff
@@ -63,9 +73,47 @@ private function createTrustedHostnames()
         });
     }
 
+    private function createDisabledDimensions()
+    {
+        return $this->makeSetting('disabled_settings', $default = [], FieldConfig::TYPE_ARRAY, function (FieldConfig $field) {
+            $field->introduction = 'Disabled Dimensions'; // TODO translate
+            $field->inlineHelp = "Disable dimensions to avoid tracking this data no matter what is sent to the tracker. This can be useful in being compliant with various privacy regulations."; // TODO: translate
+            $field->uiControl = FieldConfig::UI_CONTROL_MULTI_SELECT;
+            $field->availableValues = $this->getAvailableDimensionsToDisable();
+        });
+    }
+
     public function save()
     {
         parent::save();
         Cache::deleteTrackerCache();
     }
+
+    private function getAvailableDimensionsToDisable()
+    {
+        // TODO: translate types
+        $dimensions = [];
+        $this->addDimensions($dimensions, VisitDimension::getAllDimensions(), $type = 'Visit');
+        $this->addDimensions($dimensions, ActionDimension::getAllDimensions(), $type = 'Action');
+        $this->addDimensions($dimensions, ConversionDimension::getAllDimensions(), $type = 'Conversion');
+        return $dimensions;
+    }
+
+    /**
+     * @param string[] $dimensions
+     * @param Dimension[] $allDimensions
+     * @param $dimensionType
+     */
+    private function addDimensions(array &$dimensions, array $allDimensions, $dimensionType)
+    {
+        foreach ($allDimensions as $dimension) {
+            if (!method_exists($dimension, 'onNewVisit')
```

Comment:
instead of `method_exists` (all visit dimensions have this method) we should use eg 

```php
        $method = new \ReflectionMethod($dimension, $method);
        $declaringClass = $method->getDeclaringClass();

        0 === strpos($declaringClass->name, 'Piwik\Plugins');
```

Reply:
that's why currently no action/conversion dimensions are being shown and why userId might be shown twice